### ISSUE TITLE
 Showing Recent Errors Only /extractLocStrings:-1: bad entry in file /Users/*/*/Pods/*/*.framework/Headers/*.h (line = 365): Argument is not a literal string.

### ISSUE THREAD

**Issue by zcinclude** [Flinesoft/BartyCrouch#148]
Showing Recent Errors Only `/extractLocStrings:-1: bad entry in file /Users/*/*/Pods/*/*.framework/Headers/*.h (line = 365): Argument is not a literal string.`

here is the code in line = 365:
``` swift
#define UMLocalizedString(key,defaultValue) NSLocalizedStringWithDefaultValue(key,@"UMSocialLocalizable", [NSBundle mainBundle], defaultValue, nil)
```

**Comment by zcinclude** [comment]
Even the code is commented out, it can cause error: "Argument is not a literal string"
Like this:

  //        if (self.showManualips) {
    //            [InteractionMaster toastCautionInfoWithText:NSLocalizedString(manager.dictionaryData[@"error_msg"], manager.dictionaryData[@"error_msg"]) hideAfterDelay:2];
    //            return;
    //        }
    //    }

**Comment by Dschee** [comment]
Hey, the error is clearly stating the issue: "Argumen is not a literal string." It needs to be though in order to be discovered by `extractLocStrings`. We did configure the `quite` mode of `extractLocStrings` in #121 though to prevent such problems, I'm not sure if this applies here though. It was shipped with 4.0.1, so could you please check what version you have installed (or just upgrade and check if it's still happening?).

But I think the real issue here is the fact that the `Pods` directory is getting included in BartyCrouch searches in the first place. You shouldn't use BartyCrouch for dependencies, it should only search through your own apps code. We actually have [code in place](https://github.com/Flinesoft/BartyCrouch/blob/178bb82230f46e1a218bb76f6da8edfbb3fe2de9/Sources/BartyCrouchKit/FileHandling/FilesSearchable.swift#L20) by default to prevent searching through the `Pods` directory, but it seems not to be working for your.

Could you please provide your BartyCrouch configuration file contents? Also, please make sure to specify any parameters in there including the word `path` to the correct paths instead of using the very general `"."` default value. This will not only help fixing this issue but also make BartyCrouch run much faster.

**Comment by zcinclude** [comment]
![屏幕快照 2019-09-09 下午4 08 19](https://user-images.githubusercontent.com/12709000/64514237-cf2eb300-d31c-11e9-9386-ab204990ea0a.png)

I don't konw why bartycrouch version is 4.0.0, but brew bartycrouch version is 4.0.2.
Which one is true?

**Comment by Dschee** [comment]
That's a known issue which we already fixed. See https://github.com/Flinesoft/BartyCrouch/issues/145

**Comment by slashlos** [comment]
Hmmm, noob here, saw #145 but issue appears to persist; also a brew install
```
bartycrouch version
Version: 4.3.1
```
using this NSDataAsset extension - try for localized version first
```
if let asset = NSDataAsset.init(name: NSLocalizedString(fromAsset, comment: "")) {
	text = String.init(data: asset.data, encoding: .utf8)
}
else
if let asset = NSDataAsset.init(name: fromAsset) {
	text = String.init(data: asset.data, encoding: .utf8)
}
else
{
	return String(format: "Unable to locate asset:\n%@", fromAsset)
}
```
Is the solution to **not** expose paths containing such code to the update etc stages?

**Comment by slashlos** [comment]
So I got around this issue for me; I had an array of strings, indexed based on topic, passed to the asset handler.  The solution was to set up the source array from NSLocalizedString() calls for each literal string item - yuck., but no more error.

**Comment by billp** [comment]
The Pods directory is excluded after this fix: https://github.com/Flinesoft/BartyCrouch/pull/214
I think this issue can be closed.

**Comment by Jeehut** [comment]
Closing as fix released with 4.3.2 thanks to @billp, please report if issue still happens. 🎉
